core/lib/zksync_core/src/api_server/web3/namespaces/en.rs

```diff
@@ -15,12 +15,33 @@ impl EnNamespace {
         Self { state }
     }
 
+    pub async fn consensus_genesis_impl(&self) -> Result<Option<en::ConsensusGenesis>, Web3Error> {
+        async {
+            let Some(genesis) = self
+                .state
+                .connection_pool
+                .access_storage_tagged("api")
+                .await?
+                .consensus_dal()
+                .genesis()
+                .await?
+            else {
+                return Ok(None);
+            };
+            anyhow::Ok(Some(en::ConsensusGenesis(
+                zksync_protobuf::serde::serialize(&genesis, serde_json::value::Serializer).unwrap(),
+            )))
```

Comment:
Nit: Would it make sense to add a constructor to `ConsensusGenesis` that encapsulates this conversion? It looks weirdly low-level.

Reply:
ConsensusGenesis exists only so that zksync_types doesn't depend on zksync_consensus_roles.